contracts/Options.vy

```diff
@@ -95,8 +99,22 @@ def eth_required(amount: uint256) -> uint256:
 @view
 def _eth_required(amount: uint256) -> uint256:
     eth_per_yfi: uint256 = self._get_latest_price()
-    
-    return amount * eth_per_yfi / PRICE_DENOMINATOR / DISCOUNT
+
+    total_supply: uint256 = YFI.totalSupply()
+    total_locked: uint256 = YFI.balanceOf(VEYFI)
```

Comment:
this can be manipulated by simply transferring tokens to veYFI, not sure theres an attack possible tho, is there another function to check total locked in veYFI contract that uses state instead?

Reply:
I don't see a potential attack vector transferring tokens, but I would like to confirm the edge case where `total_locked` = `total_supply`. In this case afaik `total_locked * DISCOUNT_GRANULARITY / total_supply` would be 200 (`DISCOUNT_GRANULARITY`), and `DISCOUNT_TABLE[200]` would fail, since `DISCOUNT_TABLE` has 200 items. Please CMIIW.

Reply:
Yes, that's true. I was thinking how to break the code :)